packages/core/src/fhir-to-cda/cda-templates/components/component-of.ts

```diff
@@ -0,0 +1,119 @@
+import { Bundle, Encounter, Location, Practitioner } from "@medplum/fhirtypes";
+import { toArray } from "@metriport/shared";
+import {
+  findCompositionResource,
+  findResourceInBundle,
+  isEncounter,
+} from "../../../external/fhir/shared";
+import {
+  buildAddress,
+  buildCodeCe,
+  buildCodeCvFromCodeCe,
+  buildInstanceIdentifier,
+  buildTelecom,
+  formatDateToCdaTimestamp,
+  withoutNullFlavorObject,
+} from "../commons";
+import { placeholderOrgOid } from "../constants";
+import { AugmentedEncounter } from "./augmented-resources";
+import { createAugmentedEncounters } from "./encounters";
+import { EncompassingEncounter, ResponsibleParty } from "../../cda-types/shared-types";
+
+function findEncounterInBundle(
```

Comment:
Are we using the same encounter for the encompassingEncounter as in the Encounter section? Like is the encounter being duplicated. Not nessecarily an issue but in the case where there are multiple encounters what is the behavior here? 

Do we need some concept in the FHIR we get from customers of an encompassing encounter vs an encompassed encounter?

Reply:
Yea, so they will be duplicated in the sense that the same encounter will appear at the top level under `componentOf`, and then again under the `Encounters` section. But the way they're structured is different. 

WRT how the encompassing encounter is defined is thru the `Composition.encounter.reference`, which has a `0..1 `cardinality, so there can only be 1 enc there.

Reply:
Ok of we have 3 encounters is there a specific encounter we should be choosing as the encompassing encounter or is this being chosen At random?